src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -172,10 +181,17 @@ export class MatFormField
   set hideRequiredMarker(value: BooleanInput) {
     this._hideRequiredMarker = coerceBooleanProperty(value);
   }
-  private _hideRequiredMarker: boolean;
+  private _hideRequiredMarker = false;
 
   /** The color palette for the form-field. */
-  @Input() color: ThemePalette = 'primary';
+  @Input()
+  get color(): ThemePalette {
+    return this._color;
+  }
+  set color(value: ThemePalette) {
+    this._color = value || this._defaults?.color || 'primary';
+  }
+  private _color: ThemePalette = 'primary';
```

Comment:
Why do we need the getter/setter here? You can assign the default in the constructor.

Reply:
In case we have `[color]='undefined'` color should have a fallback value to its default.

Reply:
That's already covered by type checking.

Reply:
If we revert to just color property this test fails because `color` is `undefined` in `MatInputWithColor` https://github.com/angular/components/pull/24440/files#diff-e6d005e866330a54a6e0717ff8706c1e43a6c147765f606ed726ee7fcbd69d4cR1417
Do you prefer changing that test case instead?

Reply:
Yes, you can add use another fixture that doesn't have a `color`.

Reply:
done